middle_end/inline_and_simplify_aux.ml

```diff
@@ -65,6 +66,7 @@ module Env = struct
       inlining_stats_closure_stack =
         Inlining_stats.Closure_stack.create ();
       inlined_debuginfo = Debuginfo.none;
+      in_tail_position = true;
```

Comment:
This is not really true for toplevel expressions in Effect and Initialize_symbol:
```
let n = int_of_string "1"
```
`int_of_string` is not called in tail position here.